harness/determined/common/experimental/determined.py

```diff
@@ -404,3 +404,157 @@ def stream_trials_validation_metrics(
             trial_ids: List of trial IDs to get metrics for.
         """
         return trial._stream_validation_metrics(self._session, trial_ids)
+
+    def get_workspace_ids(
+        self,
+        session: api.Session,
+        workspace_names: List[str],
+    ) -> List[int]:
+        wnSet = set(workspace_names)
+        workspace_ids = []
+
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspacesResponse:
+            return bindings.get_GetWorkspaces(
+                session=self._session,
+                offset=offset,
+            )
+
+        resps = api.read_paginated(get_with_offset)
+        workspaces = [w for r in resps for w in r.workspaces]
+
+        for w in workspaces:
+            if w.name in wnSet:
+                if w.archived:
+                    raise Exception(f'Workspace "{w.name}" is archived.')
+                workspace_ids.append(w.id)
+        return workspace_ids
+
+    def bind_rps_to_workspaces(
```

Comment:
Per an OOB thread, there's not much in this PR that wants a unit test. Most of the functionality is in assembling parameters to be passed through to the REST API, and though we _could_ write unit tests that the requests are correct, I don't myself think that's necessary.

But nevertheless, a set of unit tests for this method could include:

* `test_bind_rp_to_workspaces_errors_on_invalid_rps`
* `test_bind_rp_to_workspaces_errors_on_invalid_workspaces`

`list_rps_bound_to_workspace` could test:
* `test_list_rps_bound_to_workspace_reads_all_pages`
* `test_list_rps_bound_to_workspace_raises_NotFound_on_invalid_workspace`
* `test_list_rps_bound_to_workspace_returns_empty_list_when_no_rps_are_bound`

Reply:
In case either is helpful, there's a [page in confluence](https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1351909582/Python+Testing) with some best practices for unit tests, and [examples already in the SDK](https://github.com/determined-ai/determined/tree/main/harness/tests/determined/common/experimental) to crib from.